magicgui/_tests/test_ qt.py

```diff
@@ -90,3 +101,66 @@ def test_set_categorical(qtbot):
     assert [w.itemText(i) for i in range(w.count())] == ["a", "b"]
     _qt.set_categorical_choices(w, (("a", 1), ("c", 3)))
     assert [w.itemText(i) for i in range(w.count())] == ["a", "c"]
+
+
+def test_magicfiledialog(qtbot):
+    """Test the MagicFileDialog class."""
+    filewidget = _qt.MagicFileDialog()
+
+    # check default values
+    assert filewidget.get_path() == Path('.')
+    assert filewidget.mode == _qt.FileDialogMode.OPTIONAL_FILE
+
+    # set the mode
+    filewidget.mode = _qt.FileDialogMode.EXISTING_FILES  # Enum input
+    assert filewidget.mode == _qt.FileDialogMode.EXISTING_FILES
+    filewidget.mode = "oPtioNal_FiLe"  # improper capitalization
+    assert filewidget.mode == _qt.FileDialogMode.OPTIONAL_FILE
+    filewidget.mode = "EXISTING_DIRECTORY"  # string input
+    assert filewidget.mode == _qt.FileDialogMode.EXISTING_DIRECTORY
+    with pytest.raises(ValueError):
+        filewidget.mode = 123  # invalid mode
+        filewidget.mode = 'invalid_string'
+
+    # set the path
+    filewidget.set_path('my/example/path/')
+    assert filewidget.get_path() == Path('my/example/path/')
+
+    filewidget.mode = _qt.FileDialogMode.EXISTING_FILES
+    filewidget.set_path(['path/one.txt', 'path/two.txt'])
+    assert filewidget.get_path() == (Path('path/one.txt'), Path('path/two.txt'))
+    filewidget.set_path(['path/3.txt, path/4.txt'])
+    assert filewidget.get_path() == (Path('path/3.txt'), Path('path/4.txt'))
+
+    with pytest.raises(TypeError):
+        filewidget.set_path(123)  # invalid type, only str/Path accepted
+
+
+@pytest.mark.skipif("sys.platform == 'darwin'")  # dialog box hangs on Mac
+@pytest.mark.parametrize("mode", [
+    ('r'),  # existing file
```

Comment:
this actually isn't surviving black formatting.
https://github.com/napari/magicgui/pull/23/checks?check_run_id=829849809#step:5:15

but it did reveal an issue in my linting test! it's (uselessly) reformatting files without raising an error. fixing in #26, but maybe try to run black locally on this?